src/electron.ts

```diff
@@ -53,26 +55,20 @@ app.on('ready', () => {
   });
 
   ipcMain.on('close', () => {
-    if (current) {
-      mainWindowState.saveState(current);
-      unregisterShortcuts();
-      current.close();
-      current = null;
-    }
+    handleWindowsOnClose();
   });
 
   ipcMain.on('closeApp', () => {
-    unregisterShortcuts();
-    if (current) {
-      mainWindowState.saveState(current);
-      current.close();
-      current = null;
-    }
+    handleWindowsOnClose();
   });
 });
 
 app.on('browser-window-created', (e: Event, win: Electron.BrowserWindow) => {
   current = win;
+  windows = new Array<Electron.BrowserWindow>()
+  if (windows.filter(w => w.id === win.id).length === 0) {
```

Comment:
This will always be true. 
Thus I am going to use `if (!windows.filter(w => w.id === win.id)[0]) { ... }` ?